server/services/actions/hpa_vpa_recommendations.py

```
@@ -0,0 +1,480 @@
+    """
+    env = (environment or "").strip() or "-"
+    return f"{env}::{(workload or '').strip()}".lower()
+
+
+def _advisory_lock_key(org_id: str, workload_key: str) -> int:
+    """Stable bigint lock key for pg_advisory_xact_lock (executor.py pattern)."""
+    digest = hashlib.sha256(f"{org_id}:{workload_key}".encode()).digest()[:7]
+    return int.from_bytes(digest, byteorder="big", signed=False) & 0x7FFFFFFFFFFFFFFF
+
+
+def _is_real_number(value: Any) -> bool:
```

Ruff (0.16.0)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)